test/ask_web/controllers/invite_controller_test.exs

```diff
@@ -1069,4 +1069,61 @@ defmodule AskWeb.InviteControllerTest do
       )
     end
   end
+
+  test "send invite to new user", %{conn: conn, user: user} do
+    Process.register(self(), :mail_target)
+    project = create_project_for_user(user)
+    code = "ABC1234"
+    level = "reader"
+    email = "user@instedd.org"
+
+    conn = get(
+      conn,
+      send_invitation_path(conn, :send_invitation, %{
+        "code" => code,
+        "level" => level,
+        "email" => email,
+        "project_id" => project.id
+      })
+    )
+
+    assert json_response(conn, 200)
+
+    email = wait_for_email()
+
+    assert email.subject == "#{user.name} has invited you to collaborate on #{project.name}."
+  end
+
+  test "send invite to existing user", %{conn: conn, user: user} do
+    Process.register(self(), :mail_target)
+    project = create_project_for_user(user)
+    code = "ABC1234"
+    level = "reader"
+    email = "user@instedd.org"
+
+    user2 = insert(:user, email: email)
+    insert(:project_membership, user_id: user2.id, project_id: project.id, level: "editor")
+
+    conn = get(
+      conn,
+      send_invitation_path(conn, :send_invitation, %{
+        "code" => code,
+        "level" => level,
+        "email" => user2.email,
+        "project_id" => project.id
+      })
+    )
+
+    assert json_response(conn, 422)["errors"]["user_id"] == ["User already in project"]
+
+    email = wait_for_email()
+
+    assert email.subject == "#{user.name} has added you as a collaborator on #{project.name}."
```

Comment:
If the user was already invited, I guess we shouldn't be sending the email - right?